source/extensions/filters/common/auth/attribute_context.cc

```diff
@@ -0,0 +1,91 @@
+#include "extensions/filters/common/auth/attribute_context.h"
+
+#include "common/http/utility.h"
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Auth {
+
+void AttributeContextUtils::setSourcePeer(envoy::service::auth::v2alpha::AttributeContext& context,
+                                          const Network::Connection& connection,
+                                          const std::string& service) {
+  setPeer(*context.mutable_source(), connection, service, false);
+}
+
+void AttributeContextUtils::setDestinationPeer(
+    envoy::service::auth::v2alpha::AttributeContext& context,
+    const Network::Connection& connection) {
+  setPeer(*context.mutable_destination(), connection, "", true);
+}
+
+void AttributeContextUtils::setPeer(envoy::service::auth::v2alpha::AttributeContext_Peer& peer,
+                                    const Network::Connection& connection,
+                                    const std::string& service, const bool local) {
+
+  auto addr = local ? connection.localAddress() : connection.remoteAddress();
+  Envoy::Network::Utility::addressToProtobufAddress(*addr, *peer.mutable_address());
+
+  if (!service.empty()) {
+    peer.set_service(service);
+  }
+
+  auto ssl = const_cast<Ssl::Connection*>(connection.ssl());
+  if (ssl) {
+    std::string principal = local ? ssl->uriSanLocalCertificate() : ssl->uriSanPeerCertificate();
+    if (principal.empty()) {
+      principal = local ? ssl->subjectLocalCertificate() : ssl->subjectPeerCertificate();
+    }
+    peer.set_principal(principal);
+  }
+}
+
+void AttributeContextUtils::setHttpRequest(
+    envoy::service::auth::v2alpha::AttributeContext& context,
+    const Envoy::Http::StreamDecoderFilterCallbacks* callbacks,
+    const Envoy::Http::HeaderMap& headers) {
+
+  auto req = context.mutable_request()->mutable_http();
+  auto sdfc = const_cast<Envoy::Http::StreamDecoderFilterCallbacks*>(callbacks);
```

Comment:
nit: Please go through and make local variables longer and easier to read. We prefer easier to read code than short names. This comment applies everywhere.